TRENZ.Lib.RazorMail.Core/Core/MailTemplateBase.cs

```diff
@@ -38,52 +43,60 @@ public void AttachFile(string filename)
         => _AttachFile(filename, ContentDisposition.Attachment);
 
     public string InlineFile(string filename, byte[] fileData)
-        => _AttachFile(filename, fileData, ContentDisposition.Inline);
+        => _AttachFile(filename, new MemoryStream(fileData), ContentDisposition.Inline);
 
     public void AttachFile(string filename, byte[] fileData)
-        => _AttachFile(filename, fileData, ContentDisposition.Attachment);
+        => _AttachFile(filename, new MemoryStream(fileData), ContentDisposition.Attachment);
 
     private string _AttachFile(string filename, ContentDisposition contentDisposition)
     {
         var contentRootPath = TempData["ContentRootPath"] as string ??
                               throw new InvalidOperationException(nameof(TempData));
 
-        var viewPath = this.Path.TrimStart('/');
+        var viewPath = Path.TrimStart('/');
 
         var parentDir = System.IO.Path.GetDirectoryName(viewPath) ?? throw new ArgumentException(nameof(this.Path));
 
         var absolutePath = System.IO.Path.Combine(contentRootPath, parentDir, filename);
 
-        return _AttachFile(filename, System.IO.File.ReadAllBytes(absolutePath), contentDisposition);
+        return _AttachFile(filename, File.OpenRead(absolutePath), contentDisposition);
     }
 
-    private string _AttachFile(string filename, byte[] fileData, ContentDisposition contentDisposition)
+    private string _AttachFile(string filename, Stream stream, ContentDisposition contentDisposition)
     {
         if (!new FileExtensionContentTypeProvider().TryGetContentType(filename, out var contentType))
             throw new NotSupportedException($"Couldn't figure out content type for {filename}");
 
-        if (!Attachments.ContainsKey(filename))
-        {
-            var attachment = new MailAttachment(fileData, filename, contentType);
-
-            switch (contentDisposition)
-            {
-                case ContentDisposition.Inline:
-                    attachment.ContentId = filename;
-                    attachment.Inline = true;
+        var cid = $"cid:{filename}";
+        if (Attachments.ContainsKey(filename))

```

Comment:
possibly useful, but also possibly not what the user intended; what if they want to attach two files with the same name? (and definitely scope creep)

Reply:
I didn't even know this was possible with mail attachments. I think I didn't change the logic, so this wasn't possible before either. The only thing I changed was the nesting to improve readability.

Reply:
My bad.

This behavior is probably wrong; see #21.